src/richie/apps/search/indexers/courses.py

```diff
@@ -46,6 +49,36 @@ class CoursesIndexer:
         },
     }
 
+    scripts = {
+        # The ordering process first splits the courses into four groups, with further ordering
+        # inside each one of those groups.
+        "sort_list": {
```

Comment:
Maybe add explanation with a mini graph explaining that each group gets a timestamp in n * maxdate?
Let me try to draw it "NS daterange style":
```
    |------------|11111111111|------------222222222222|------------333333333333|------------444444444444|
mindate    now            maxdate                     2*maxdate                    3*maxdate                    4*maxdate
```

Reply:
I added a graph sorta like yours